### DISCUSSION TITLE
Adding the ability to adjust Booster tube use.

### DISCUSSION THREAD

**Discussion by vincemic** [gpstar81/GPStar-proton-pack#411] [Ideas]
I want to add the ability to have booster tube and fan to turn on when pack turns on, instead of during a overheat. I used the two connections for the atomizer and led rope effects as shown in this video. https://youtube.com/shorts/v15teEuBYOc?si=h6fKe0sZELAou3gr . I can do the modifications myself if you are willing to entertain a pull request for the feature.

**Comment by nomakewan** [comment]
The issue here will be the menu. The vast majority of our users use the booster tube smoke as part of the overheat sequence.

How would you propose implementing user choice for your configuration change?

**Reply by vincemic** [reply]
I would like a simple choice on both the booster tube smoke and fan. The choice would be Turn On with values "With Overheat" or "Pack Power On" or something along those lines.

**Reply by nomakewan** [reply]
At the moment there just isn't enough room for such a thing: https://github.com/gpstar81/GPStar-proton-pack/blob/main/OPERATION_EEPROM.md

Unless you have a creative solution. Perhaps once we re-think the menus for 6.0 there's a chance such an implementation could be looked at, but for now I don't see it.

I'll keep this discussion open however; 6.0's changes are not set in stone, and it will be quite a while before the first firmware gets released. There is always time for improvement and feedback.

**Reply by vincemic** [reply]
I am sorry, when you said room, did you mean EEPROM, webpage storage, development time, or something else? If its EEPROM storage, I noticed the config settings seem to be uniformly all 8 bit values. Many of the config values can be something smaller. I know there is a lot of things to consider, like migration of configuration between versions and such.  I didn't expect anything to make the next release for sure.

**Reply by nomakewan** [reply]
By "room" I literally meant "space in the menu system." As in, a place to actually put the option in the user-facing part of the menu.

There are five levels to the EEPROM Configuration Menu. With the exception of Level 1 (where option 5 must be split between resetting to defaults and saving the current settings) all levels have five possible options, each with two sub-options (one for the Intensify switch, one for the Barrel Wing Button). This gives the EEPROM Configuration Menu a total of 48 possible configuration toggles.

As of now, all of those toggles are in use. That is what I mean by "room".

**Comment by vincemic** [comment]
I get it now, yes.
